hyperspy/io.py

```diff
@@ -771,6 +772,14 @@ def save(filename, signal, overwrite=None, **kwds):
             f"Please try one of {strlist2enumeration(yes_we_can)}"
         )
 
+    if writer.non_uniform_axis is False and nua is False:
+        yes_we_can = [plugin.format_name for plugin in io_plugins
+                      if plugin.non_uniform_axis is True]
+        raise OSError("Writing to this format is not supported for "
```

Comment:
OR `IOError` like for  the "This file format does not support this data." exception just before? Would make sense to use the same kind of error for both I would say. What do you think?

Reply:
Initially, I thought the same, but `IOError` is an alias for `OSError` and in this case, I prefer `TypeError` because we are trying to save a type of axes which is not supported, which I think is closer to the description in https://docs.python.org/3/library/exceptions.html#TypeError than https://docs.python.org/3/library/exceptions.html#OSError

> Changed in version 3.3: EnvironmentError, IOError, WindowsError, socket.error, select.error and mmap.error have been merged into OSError, and the constructor may return a subclass.

from https://docs.python.org/3/library/exceptions.html#OSError.filename2

Reply:
OK, but then I'll change the IOError above as well.